lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Backup.psm1

```diff
@@ -0,0 +1,24 @@
+# Copyright (c): 2018, Dag Wieers (@dagwieers) <dag@wieers.com>
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+Set-StrictMode -Version 2.0
+$ErrorActionPreference = "Stop"
+
+Function Backup-File($path, $obj=@{}, $check_mode=$false) {
+<#
+    .SYNOPSIS
+    Helper function to make a backup of a file.
+    .EXAMPLE
+    Backup-File -path $path -obj $result
+#>
+    $backup_path = "$path.$pid." + [DateTime]::Now.ToString("yyyyMMdd-HHmmss") + ".bak";
+    Try {
+        Copy-Item -Path $path -Destination $backup_path -WhatIf:$check_mode
+    } Catch {
+        Fail-Json -obj $obj -message "Failed to create backup file '$backup_path' from '$path'. ($($_.Exception.Message))"
```

Comment:
Calling `Fail-Json` in a module util isn't the best practice, there's no way for a module to handle these failures which may be needed in the future. I would just do `throw "Failed to create backup file '$backup_path' from '$path'. ($($_.Exception.Message))"`. The new exception handling will make sure this message stays sane even if the module doesn't catch it in it's own code.

Reply:
I could also remove the try-catch statements altogether, but I am not sure if the resulting error would be clear enough to what the cause of the issue is. (i.e. it is backup-related).

Reply:
Did what you suggested anyway :-)